core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -360,7 +360,7 @@ protected void doSample() {
                     try {
                         // its a listed node, light connect to it...
                         logger.trace("connecting to listed node (light) [{}]", listedNode);
```

Comment:
I am not sure how those flips from non-light to light and vice-versa happened, but I will fix. Thanks for catching.